tests/tests/test-staking/test-staking-locks.ts

```diff
@@ -539,24 +533,40 @@ describeDevMoonbeam("Staking - Locks - bottom and top delegations", (context) =>
   });
 
   it("should be set for bottom and top list delegators", async function () {
-    await expectOk(
-      context.createBlock(
-        [...topDelegators].map((account, i) =>
-          context.polkadotApi.tx.parachainStaking
-            .delegate(alith.address, MIN_GLMR_DELEGATOR + 1n * GLMR, i + 1, 1)
-            .signAsync(account)
-        )
-      )
-    );
-    await expectOk(
-      context.createBlock(
-        [...bottomDelegators].map((account, i) =>
-          context.polkadotApi.tx.parachainStaking
-            .delegate(alith.address, MIN_GLMR_DELEGATOR, topDelegators.length + i + 1, 1)
-            .signAsync(account)
-        )
-      )
-    );
+    this.timeout(1000000)
```

Comment:
This test is very slow now because it is creating a block for each single extrinsic.

As pointed out earlier, having the same (0) priority for each txn seems to prevent block production from filling all txns in one block.

TODO: find workaround so this test isn't so slow

Reply:
Shouldn't the block production take into account the tx weight and only then the priority? I'm confused why setting a priority would lead to this behavior

Reply:
Shouldn't the block production take into account the tx weight and only then the priority? I'm confused why setting a priority would lead to this behavior

Reply:
I think the problem here (still some guesswork) is that my change causes Substrate txns with no tip to have 0 priority (note this would be the same for Eth txns right now with no `max_priority_fee_per_gas`). So they all end up in the pool with an identical priority and this is causing perf issues.

Obviously this needs a closer look :)